### PR TITLE
fix(core): handle missing translation issue in document titles

### PR DESCRIPTION
**Pull request by aliemir**
Update default document title generation to use the fallback title when `i18n`'s `translate` function returns the `key` value.

### Test plan (required)

```sh
Test Suites: 126 passed, 126 total
Tests:       1001 passed, 1001 total
Snapshots:   18 passed, 18 total
Time:        82.581 s, estimated 90 s
Ran all test suites.
```

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/core/src/definitions/helpers/safe-translate/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { safeTranslate } from ".";
```
Comment (BatuhanW) [inline packages/core/src/definitions/helpers/safe-translate/index.test.ts:1]: I think it's better to follow naming convention for all other tests(except inferencer tests 😄)

`index.spec.ts`

File: packages/core/src/definitions/helpers/safe-translate/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { useTranslate } from "@hooks/translate";
```
Comment (BatuhanW) [inline packages/core/src/definitions/helpers/safe-translate/index.ts:1]: Can we use like below?
```suggestion
import type { useTranslate } from "@hooks/translate";
```

File: packages/core/src/definitions/helpers/safe-translate/index.test.ts (hunk 1 above)
Comment (aliemir) [reply packages/core/src/definitions/helpers/safe-translate/index.test.ts]: Done ✅

File: packages/core/src/definitions/helpers/safe-translate/index.ts (hunk 2 above)
Comment (aliemir) [reply packages/core/src/definitions/helpers/safe-translate/index.ts]: Not needed actually, but updated them anyways.

**Review by BatuhanW — APPROVED** [review]

**Pull request merged**